tsdb/engine/tsm1/compact.go

```diff
@@ -553,45 +553,49 @@ func (c *Compactor) write(path string, iter KeyIterator) error {
 	if err != nil {
 		return err
 	}
-	defer w.Close()
```

Comment:
Instead of replacing this with a large, anonymous function. What do you think about naming the return parameter and doing this:
```
defer func() { closeErr := w.Close(); if err == nil { err = closeErr } }()
```

The current code also doesn't close the file if an error occurred, potentially holding on to filesystem space until the program closes.